inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -144,7 +144,7 @@ def _handle_app_source_command(
 
         application_source_manager.add(
             ApplicationAddSourceParameters(
-                file_name=repo_filename,
+                source_list_file_name=repo_filename,
                 gpg_key_name=gpg_key_name,
                 gpg_key_uri=gpg_key_uri,
```

Comment:
**Suggestion:** Refactor the repeated code for extracting gpg_key_uri and gpg_key_name into a separate function to improve code maintainability and reduce duplication. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
def extract_gpg_keys(parsed_head, action):
    gpg_key_uri = None
    gpg_key_name = None
    try:
        gpg_keys = parsed_head.get_children(f"applicationSource/{action}/gpg")
        gpg_key_uri = gpg_keys["uri"]
        gpg_key_name = gpg_keys["keyname"]
    except XmlException:
        logger.info(f"Optional GPG key parameters not present in {action} manifest")
    return gpg_key_uri, gpg_key_name

gpg_key_uri, gpg_key_name = extract_gpg_keys(parsed_head, "add")
```